time/ptp.md

```diff
@@ -0,0 +1,218 @@
+
+User configures linuxptp services and apply rules in `NodePTPCfg` custom
+resource. Linuxptp services are configured via `profile` section in `NodePTPCfg`
+and apply rules are configured via `recommend` section in `NodePTPCfg`.
+Multiple `NodePTPCfg` custom resources can be created, PTP DaemonSet will merge
+all the `NodePTPCfg` custom resources according to recommend definition and
+apply one profile that best matches with the node.
+
+Upon receiving a notification event of `NodePTPCfg` creation or update,
+PTP Daemon verify the correctness of `NodePTPCfg` custom resource and
+apply the selected profile to each node. One profile at most will be
+applied to each node.
+
+### User Stories [optional]
+
+#### Virtual Radio Access Networks
+
+NFV vRAN workloads can run on OpenShift baremetal cluster.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The proposal introduces PTP as Tech Preview.
+
+#### Node PTP Devices
+
+Hardware PTP capability on network interface is detected with the following
+command:
+
+`ethtool -T <network-interface-name>`
+
+This command shows whether a MAC supports hardware time stamping. Only devices
+that contain below flags are listed as PTP capable device.
+
+```
+SOF_TIMESTAMPING_TX_HARDWARE
+SOF_TIMESTAMPING_RX_HARDWARE
+SOF_TIMESTAMPING_RAW_HARDWARE
+```
+
+#### PTP Clock Types
+
+Clocks synchronized by PTP are organized under a hierachy of master-slave type.
+The slaves are synchronized to their masters who can themselves be slaves to
+other masters. When a clock has only one port, it can be master or slave, this
+type of clock is called an `Ordinary Clock`, A clock with multiple ports can be
+master on one port and slave on another, this type of clock is called a
+`Boundary Clock`. The PTP Clock type supported in this proposal is
+`Ordinary Clock` and only one network interface on each node can be configured
+for PTP.
+
+#### Node PTP Config
+
+Although multiple `NodePTPCfg` custom resources can be defined in the cluster,
+only one profile can be applied to a specific node. Each profile contains
+combination of `Interface`,  `ptp4lOpts` and `phc2sysOpts`, one clock at most
+can be configured on each node, this excludes the possibility to configure an
+OpenShift node to be a PTP `Boundary Clock` node.
+
+#### PTP Network Transport Mode
+
+PTP protocol supports three transport modes: IEEE 802.3, UDP IPv4 and UDP IPv6.
+When using UDP network transport mode, PTP device requires IP address be
+configured, however this is not supported by PTP daemon in this proposal, user
+will need to configure it manually. `NodePTPCfg` might be extended in the future
+to support necessary network configurations such as creating a bond interface
+for PTP redundancy, assigning IP address for UDP transport mode.
+
+#### PTP4L configuration file
+
+Default ptp4l configuration file (ptp4l.conf) will be used when starting ptp4l
+service, user doesn't need to specify ptp4l.conf in `ptp4lOpts`. The content
```

Comment:
As I understand, the daemonset will use the default configuration from ptp4l.conf and the user can overwrite some of the options via ptp4lOpts. It would be worth to mention where this options will be stored (like a configMap or in /etc/sysconfig/ptp4l).

Reply:
Correct! 
updated the doc to add:
1) ptp4l.conf is installed in ptp daemon image by default under `/etc/ptp4l.conf`
2) `-f ptp4l.conf` will be automatically appended to `ptp4lOpts` by PTP daemon
3) some of `ptp4l.conf` configs can be overwritten by specifying certain options in `ptp4lOpts`.

Reply:
LGTM